src/layout/BaseLayout.tsx

```diff
@@ -1,14 +1,24 @@
+import React from 'react';
+
 import { Outlet } from 'react-router-dom';
 
 import { Box } from '@mui/material';
 
 import { Header, Sidebar } from '@containers';
 import { LayoutProps } from '@types';
 
-export const BaseLayout = ({ showSideBar }: LayoutProps) => (
-    <Box display="flex">
-        <Header />
-        {showSideBar && <Sidebar />}
-        <Outlet />
-    </Box>
-);
+export const BaseLayout = ({ showSideBar }: LayoutProps) => {
+    const [isSidebarOpen, setIsSidebarOpen] = React.useState(false);
+    const toggleSidebar = () => {
+        setIsSidebarOpen(!isSidebarOpen);
+    };
+    return (
+        <Box display="flex">
+            <Header onMenuClick={toggleSidebar} />
+            {showSideBar && (
+                <Sidebar open={isSidebarOpen} onClose={toggleSidebar} />
+            )}
```

Comment:
check if we can remove this
```suggestion
            
                <Sidebar open={isSidebarOpen} onClose={toggleSidebar} />
            
```